components/scream/src/share/field/field.hpp

```diff
@@ -471,63 +471,58 @@ sync_to_dev () const {
   EKAT_REQUIRE_MSG(layout==layout_src,
        "ERROR: Unable to copy field " + field_src.get_header().get_identifier().name() + 
-          " to field " + get_header().get_identifier().name() + " during initialization.  Layouts don't match.");
+          " to field " + get_header().get_identifier().name() + ".  Layouts don't match.");
   const auto  rank = layout.rank();
   // Note: we can't just do a deep copy on get_view<HD>(), since this
   //       field might be a subfield of another. So check the header
   //       first, to see if we have a parent. If not, deep copy is fine.
   //       If we do, we need to get the reshaped view first.
   const auto parent = get_header().get_parent();
-  if (parent.lock()==nullptr) {
-    // No parent. Deep copying to get_view<HD>() is safe.
-    Kokkos::deep_copy (get_view<HD>(),field_src.get_view<HD>());
-  } else {
-    // We have a parent. We only want to set *this* field to value,
-    // not the rest of the parent field. We need the reshaped view
-    switch (rank) {
-      case 1:
-        {
-          auto v     = get_reshaped_view<RT*,HD>();
-          auto v_src = field_src.get_reshaped_view<RT*,HD>();
-          Kokkos::deep_copy(v,v_src);
-        }
-        break;
-      case 2:
-        {
-          auto v     = get_reshaped_view<RT**,HD>();
-          auto v_src = field_src.get_reshaped_view<RT**,HD>();
-          Kokkos::deep_copy(v,v_src);
-        }
-        break;
-      case 3:
-        {
-          auto v     = get_reshaped_view<RT***,HD>();
-          auto v_src = field_src.get_reshaped_view<RT***,HD>();
-          Kokkos::deep_copy(v,v_src);
-        }
-        break;
-      case 4:
-        {
-          auto v     = get_reshaped_view<RT****,HD>();
-          auto v_src = field_src.get_reshaped_view<RT****,HD>();
-          Kokkos::deep_copy(v,v_src);
-        }
-        break;
-      default:
-        EKAT_ERROR_MSG ("Error! Unsupported field rank in 'set_value'.\n");
-    }
+  // We have a parent. We only want to set *this* field to value,
+  // not the rest of the parent field. We need the reshaped view
+  switch (rank) {
+    case 1:
+      {
+        auto v     = get_reshaped_view<RT*,HD>();
+        auto v_src = field_src.get_reshaped_view<RT*,HD>();
+        Kokkos::deep_copy(v,v_src);
+      }
+      break;
+    case 2:
+      {
+        auto v     = get_reshaped_view<RT**,HD>();
+        auto v_src = field_src.get_reshaped_view<RT**,HD>();
+        Kokkos::deep_copy(v,v_src);
+      }
+      break;
+    case 3:
+      {
+        auto v     = get_reshaped_view<RT***,HD>();
+        auto v_src = field_src.get_reshaped_view<RT***,HD>();
+        Kokkos::deep_copy(v,v_src);
+      }
+      break;
+    case 4:
+      {
+        auto v     = get_reshaped_view<RT****,HD>();
+        auto v_src = field_src.get_reshaped_view<RT****,HD>();
+        Kokkos::deep_copy(v,v_src);
+      }
+      break;
+    default:
```

Comment:
Would you mind adding also the rank-5 case? The rank-4 is probably unused on physics grid, and meant for dyn grid. But a 3d vector quantity on dyn grid would have rank 5 (EL,CMP,GP,GP,LEV). Since it's almost the same code, we might as well add it.

Reply:
done.